python/huggingfaceserver/huggingfaceserver/test_model.py

```diff
@@ -257,12 +257,13 @@ async def test_bloom_completion(bloom_model: HuggingfaceGenerativeModel):
         prompt="Hello, my dog is cute",
         stream=False,
         echo=True,
+        max_tokens=100,  # bloom doesn't have any field specifying context length. Our implementation would default to 2048. Testing with something longer than HF's default max_length of 20
```

Comment:
Can we can a separate test for max_tokens 100?

Reply:
the chat completion test seems to be failing

Reply:
just did a test, seems like for `CreateChatCompletionRequest` it defaults `max_tokens` to None

```python
>>> b = CreateChatCompletionRequest(model="test", messages=[{"role": "system", "content": "testtestst"}])
>>> b
CreateChatCompletionRequest(messages=[ChatCompletionRequestSystemMessage(content='testtestst', role='system', name=None)], model='test', frequency_penalty=0, logit_bias=None, logprobs=False, top_logprobs=None, max_tokens=None, n=1, presence_penalty=0, response_format=None, seed=None, stop=None, stream=False, temperature=1, top_p=1, tools=None, tool_choice=None, user=None, function_call=None, functions=None)
```

Reply:
Interestingly chat completion defaults max_token to None while completion request defaults to 16

https://github.com/kserve/kserve/blob/16c90170721e6c2ed121204fa848d41e9f4f4e96/python/kserve/kserve/protocol/rest/openai/types/openapi.py#L2643

Reply:
I think openai's expectation is that chat complete models are supposed to end when they generate <eos_token> (or equivalent) because they are trained to finish their responses.
So things should be better if we use a chat/instruct model for this test?
Something like [DialoGPT](https://huggingface.co/microsoft/DialoGPT-small) which is small yet generates dialogues that are limited in length.
The responses can be non-sensical but there'll be a response

Reply:
[tinyllama(&chat)](huggingface.co/tinyllama/tinyllama-1.1b-chat-v1.0) is a good model but its very big in comparison to the above one.
And 2x as big as bloom.